superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -588,79 +592,81 @@ export default function nvd3Vis(slice, payload) {
-        chart.xAxis.axisLabel(fd.x_axis_label).axisLabelDistance(distance);
+        chart.xAxis.axisLabel(xAxisLabel).axisLabelDistance(distance);
       }
 
-      if (fd.y_axis_label && fd.y_axis_label !== '' && chart.yAxis) {
+      if (yAxisLabel && yAxisLabel !== '' && chart.yAxis) {
         let distance = 0;
-        if (margins.left && !isNaN(margins.left)) {
+        if (margins.left && !Number.isNaN(margins.left)) {
           distance = margins.left - 70;
         }
-        chart.yAxis.axisLabel(fd.y_axis_label).axisLabelDistance(distance);
+        chart.yAxis.axisLabel(yAxisLabel).axisLabelDistance(distance);
       }
 
-      const annotationLayers = (slice.formData.annotation_layers || []).filter(x => x.show);
-      if (isTimeSeries && annotationLayers && slice.annotationData) {
+      if (isTimeSeries && annotationData && annotationLayers.length > 0) {
         // Time series annotations add additional data
         const timeSeriesAnnotations = annotationLayers
-          .filter(a => a.annotationType === AnnotationTypes.TIME_SERIES).reduce((bushel, a) =>
-        bushel.concat((slice.annotationData[a.name] || []).map((series) => {
-          if (!series) {
-            return {};
-          }
-          const key = Array.isArray(series.key) ?
-            `${a.name}, ${series.key.join(', ')}` : `${a.name}, ${series.key}`;
-          return {
-            ...series,
-            key,
-            color: a.color,
-            strokeWidth: a.width,
-            classed: `${a.opacity} ${a.style} nv-timeseries-annotation-layer showMarkers${a.showMarkers} hideLine${a.hideLine}`,
-          };
-        })), []);
+          .filter(layer => layer.show)
+          .filter(layer => layer.annotationType === AnnotationTypes.TIME_SERIES)
+          .reduce((bushel, a) =>
+            bushel.concat((annotationData[a.name] || []).map((series) => {
+              if (!series) {
+                return {};
+              }
+              const key = Array.isArray(series.key) ?
+                `${a.name}, ${series.key.join(', ')}` : `${a.name}, ${series.key}`;
+              return {
+                ...series,
+                key,
+                color: a.color,
+                strokeWidth: a.width,
+                classed: `${a.opacity} ${a.style} nv-timeseries-annotation-layer showMarkers${a.showMarkers} hideLine${a.hideLine}`,
+              };
+            })), []);
         data.push(...timeSeriesAnnotations);
       }
 
       // render chart
       svg
-      .datum(data)
-      .transition().duration(500)
-      .attr('height', height)
-      .attr('width', width)
-      .call(chart);
+        .datum(data)
+        .transition().duration(500)
+        .attr('width', width)
+        .attr('height', height)
+        .call(chart);
 
       // on scroll, hide tooltips. throttle to only 4x/second.
-      $(window).scroll(throttle(hideTooltips, 250));
+      window.addEventListener('scroll', throttle(hideTooltips, 250));
```

Comment:
wow, didn't know this was there. wonder if this is necessary since the tooltips go away quickly anyway.

Reply:
yeah, kinda surprise, but I will not touch this for now.

Reply:
I forgot the exact reason it's there, but would dig as deep as needed in `bit blame` to understand the reason before/if removing it.

Reply:
from #3518 by chart annotation feature.